### PR TITLE
ELK integration

### PR DESCRIPTION
**Pull request by shaik80**
Signed-off-by: shaik80 <shaik.mudassir@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Form added for ELK webhook integration
Test connection with integration
Save integration with integration

### :chains: Related Resources

#5577

### :+1: Definition of Done
 Should perform all the tasks for ELK Integration 
- Save 
- Update name and URL
- Test connection before and after saving destination
- Delete
- Enable and Disable destination 

### :athletic_shoe: How to Build and Test the Change
```
STEP 1

inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

make serve

To run cypress test
cd e2e
npm run cypress:local

Type feat to enable data-feed
Navigate to settings/data-feeds
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/47217471/133208627-56f2c8b6-aa28-4ed8-933b-4cd281c76d5f.mov

### CODE REVIEW

**Review by vivekshankar1 — CHANGES_REQUESTED** [review]
Please add unit test cases for saveDestination, testDestination and validateForm for elk. In cypress please add test case for test failure

File: components/automate-ui/src/app/pages/data-feed-create/data-feed-create.component.html [hunk 1]
```diff
@@ -27,7 +27,7 @@ <h2 class="bolder-font">Webhook Integration</h2>
             <div class="container image-footer">
               <h3 class="bolder-font">{{item.name}}</h3> 
             </div>
-            <div class="overlay" *ngIf="item.name !== 'ServiceNow' && item.name !== 'Splunk'">
+            <div class="overlay" *ngIf="item.name !== 'ServiceNow' && item.name !== 'Splunk' && item.name !== 'ELK'">
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/pages/data-feed-create/data-feed-create.component.html:30]: can we have this validation in a function ?

File: components/automate-ui/src/app/pages/data-feed-create/data-feed-create.component.spec.ts [hunk 2]
```diff
@@ -179,7 +191,20 @@ describe('DataFeedCreateComponent', () => {
...
+      component.createForm.controls['token'].setValue('any');
+      component.slidePanel();
+      component.name = jasmine.createSpyObj('name', ['nativeElement']);
+      component.name.nativeElement = { focus: () => { }};
+      component.selectIntegration('ELK');
+      expect(component.createForm.controls['name'].value).toBe(null);
+      expect(component.createForm.controls['url'].value).toBe(null);
+      expect(component.createForm.controls['tokenType'].value).toBe('Bearer');
+      expect(component.createForm.controls['token'].value).toBe(null);
+    });
```
Comment (MadhviAtroliya) [inline components/automate-ui/src/app/pages/data-feed-create/data-feed-create.component.spec.ts:207]: Put some spaces between two test cases.

File: components/automate-ui/src/app/pages/data-feed-create/data-feed-create.component.ts [hunk 3]
```diff
@@ -203,6 +203,13 @@ export class DataFeedCreateComponent {
         this.integrationSelected = true;
         break;
       }
+      case WebhookIntegrationTypes.ELK_KIBANA: {
+        this.showFieldWebhook();
+        this.authSelected = AuthTypes.USERNAMEANDPASSWORD;
```
Comment (MadhviAtroliya) [inline components/automate-ui/src/app/pages/data-feed-create/data-feed-create.component.ts:208]: In story ELK page open with default authentication as Access token, is this ok to show with Username and password?

File: e2e/cypress/integration/ui/settings/data-feed/data-feed-create.spec.ts [hunk 4]
```diff
@@ -86,8 +86,23 @@ describe('chef datafeed', () => {
...
+      cy.get('[data-cy=select-username-password]').click();
+      cy.get('[data-cy=add-username]').type(username);
+      cy.get('[data-cy=add-password]').type(password);
+      cy.get('[data-cy=add-button]').click();
+      cy.get('app-notification.info').should('be.visible');
+      cy.get('app-notification.info chef-icon').click();
+      cy.contains('Data Feeds').click();
+      cy.get('chef-table chef-tbody chef-td').contains('cytest' + date).should('exist');
+    });
+    it('create data feed error splunk', () => {
```
Comment (MadhviAtroliya) [inline e2e/cypress/integration/ui/settings/data-feed/data-feed-create.spec.ts:105]: Give space between two different test cases

**Review by MadhviAtroliya — APPROVED** [review]

**Review by shivangi-singh1 — APPROVED** [review]

**Review by vivekshankar1 — APPROVED** [review]
Thanks shaik, its working properly

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**